Code/Framework/AzFramework/AzFramework/Asset/UnusableAssetTypes.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Asset/AssetCommon.h>
+
+namespace AzFramework
+{
+    // A list of editor unusuable asset types that are filtered out through a toggle switch in the asset browser 
+
+    inline constexpr AZ::Data::AssetType AzslOutcomeAssetType{ "{6977AEB1-17AD-4992-957B-23BB2E85B18B}" };
+    inline constexpr AZ::Data::AssetType ModelLodAssetType{ "{65B5A801-B9B9-4160-9CB4-D40DAA50B15C}" };
+    inline constexpr AZ::Data::AssetType ImageMipChainAssetType{ "{CB403C8A-6982-4C9F-8090-78C9C36FBEDB}" };
+    inline constexpr AZ::Data::AssetType BufferAssetType{ "{F6C5EA8A-1DB3-456E-B970-B6E2AB262AED}" };
+    inline constexpr AZ::Data::AssetType ShaderVariantAssetType{ "{51BED815-36D8-410E-90F0-1FA9FF765FBA}" };
```

Comment:
We can actually data drive this via the Settings Registry.

Instead of having developers hard code this list of asset types to filter from the asset browser, instead the user can provide a .setreg file that reads data from a specific json path such as `/O3DE/AssetBrowser/AssetTypeUuidExcludes` and use that as a list of asset types to filter.

For example these Uuids can be moved to a setreg file that would get merged into the Settings Registry when Atom RPI gem loads.

### `<atom-rpi0gem-path>/Registry/asset_browser.setreg`
```json
{
    "O3DE": {
        "AssetBrowser": {
            "AssetTypeUuidExcludes": {
                "AzslOutcomeAsset":  "{6977AEB1-17AD-4992-957B-23BB2E85B18B}",
                "ModelLodAsset": "{65B5A801-B9B9-4160-9CB4-D40DAA50B15C}",
                "ImageMipChainAsset": "{CB403C8A-6982-4C9F-8090-78C9C36FBEDB}", 
                "BufferAsset": "{F6C5EA8A-1DB3-456E-B970-B6E2AB262AED}", 
                "ShaderVariantAsset": "{51BED815-36D8-410E-90F0-1FA9FF765FBA}"
            }
        }
    }
}
```

Those settings would merge with any other `/O3DE/AssetBrowser/AssetTypeUuidExcludes` settings that are loaded from other JSON files using the JSON Merge Patch functionality in the Settings Registry.

Next the Settings Registry [Visitor API](https://www.o3de.org/docs/user-guide/settings/developer-api/#visitor-api) can be used to read every Asset Type Uuid string from the Settings Registry

Reply:
Done! Updated to use the Settings Registry instead